readthedocs/rtd_tests/tests/test_redirects.py

```diff
@@ -175,6 +191,21 @@ def test_redirect_page(self):
         self.assertEqual(
             r['Location'], 'http://pip.readthedocs.org/en/latest/tutorial/install.html')
 
+    @override_settings(USE_SUBDOMAIN=True)
+    def test_redirect_with_query_params(self):
+        Redirect.objects.create(
+            project=self.pip, redirect_type='page',
+            from_url='/install.html', to_url='/tutorial/install.html'
+        )
+        r = self.client.get(
+            '/install.html?foo=bar', HTTP_HOST='pip.readthedocs.org'
+        )
+        self.assertEqual(r.status_code, 302)
+        self.assertEqual(
+            r['Location'],
+            'http://pip.readthedocs.org/en/latest/tutorial/install.html?foo=bar'
```

Comment:
A test with the `#fragment` part would be awesome to add here, like: `/install.html?foo=bar#section`

Reply:
The original issue has `#build-flags` on the URL.

Reply:
Same as https://github.com/rtfd/readthedocs.org/pull/5081#discussion_r246420846